### PR TITLE
Selenium chrome spike sprint20

### PR DESCRIPTION
**Pull request by Zsuark**
PT Story: https://www.pivotaltracker.com/story/show/148950573

Changes proposed in this pull request:
1. Adds in `@selenium` (headless) and `@selenium_browser` (displayed) chrome browser
2. Adds in `/^I confirm the popup$/` to the step definitions. NOTE: `/^I confirm popup$/` remains for poltergeist use

Ready for review

### CODE REVIEW

**Review by RobertCram — APPROVED** [review]
It seems the Semaphore error is just a fluke (I tried a copy of the same PR and it passed). If you remove the #TODO comment in the gem file and commit again, I'm fairly sure the build will pass.

Headless chrome not recognising alerts seems to be a well known problem. There is a bug report here:

https://bugs.chromium.org/p/chromium/issues/detail?id=696439

And someone trying to workaround it in Capybara here:

https://github.com/teamcapybara/capybara/commit/12c10059709346bfeadb2cea18f510174d842303

File: features/step_definitions/basic_steps.rb [hunk 1]
```diff
@@ -13,6 +13,11 @@
   end
 end
 
+And /^I confirm the popup$/ do
+  # This is for CHROME/Selenium (not headless)
+  page.driver.browser.switch_to.alert.accept
+end
+
```
Comment (patmbolger) [inline features/step_definitions/basic_steps.rb:20]: Recommend consolidating this and the prior method to this:

```
When /^I confirm popup$/ do

  if Capybara.current_driver == :poltergeist
    using_wait_time 3 do
      page.driver.accept_modal(:confirm)
    end
  elsif Capybara.current_driver == :selenium_browser
    page.driver.browser.switch_to.alert.accept
  else
    raise 'step not configured for current browser driver'
  end

end
```

File: features/support/env.rb [hunk 2]
```diff
@@ -39,5 +39,31 @@ def i18n_content(content, locale=I18n.locale)
...
+# Use @selenium_browser
+Capybara.register_driver :selenium_browser do |app|
+  Capybara::Selenium::Driver.new(
+    app,
+    browser: :chrome
+  )
+end
+
+
+# TODO: Needed?
```
Comment (patmbolger) [inline features/support/env.rb:67]: No, we don't need line 69.  Our `hooks.rb` handles this.

**Review by patmbolger — CHANGES_REQUESTED** [review]
Please see comments.

**Comment by patmbolger** [comment]
Closing this PR and creating a new PR to replace.

**Pull request closed without merging**